packages/pwa/app/components/footer/index.jsx

```diff
@@ -135,8 +135,8 @@ const Footer = ({...otherProps}) => {
                                 {...styles.localeDropdown}
                             >
                                 {SUPPORTED_LOCALES.map((locale) => (
-                                    <option key={locale} value={locale}>
-                                        {intl.formatMessage(defaultLocaleMessages[locale])}
+                                    <option key={locale.id} value={locale.id}>
```

Comment:
Try simulating a slow network and you'll see that after selecting a new locale from the dropdown you won't see it being selected. To fix this you can use a local state for the locale value like this:
```
const [locale, setLocale] = useState(intl.locale)
```

then use the locale value as the select value, and also setLocale(target.value) in the onChange function.